File: Models/Icarus/Form/Select.cs
﻿using ICARUS.Models.Icarus;
using ICARUS.Models.Icarus.Elements;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Web.Mvc;
namespace ICARUS.Models {
    /// <summary>
    /// A SELECT Element (INPUT, SELECT etc)
    /// </summary>
    public class Select : Container {
        /// <summary>
        /// Generic SELECT Constructor for Entity
        /// </summary>
        public Select() : base("SELECT", new MODEL() {
            label = "SELECT"
        }) {
            this.className = "SELECT";
        }
        /// <summary>
        /// Constructs an SELECT with the given id and label attributes already set
        /// </summary>
        /// <param name="formPost"></param>
        public Select(FormPost formPost) : base("SELECT", formPost) {

        }
    }
}

File: Scripts/src/icarus/model/EL/select/SELECT.js
/** @module */
import EL, { MODEL } from '../EL.js';
/** A SELECT element */
export default class SELECT extends EL {
    /** A Select Element Constructor
        @param {EL} node Node
        @param {MODEL} model Model
    */
	constructor(node, model) {
		super(node, 'SELECT', model);
	}
}